### PR TITLE
Add ApiListener#tls_handshake_timeout option

### PR DESCRIPTION
**Pull request by dnsmichi**
This allows to specify the previously hardcoded
timeout of 10s.

refs #6517

### CODE REVIEW

**Comment by dnsmichi** [comment]
Fixed the Windows build and some type casts with double.

File: lib/base/tlsstream.cpp [hunk 1]
```diff
@@ -315,14 +317,13 @@ void TlsStream::Handshake()
 	m_CurrentAction = TlsActionHandshake;
 	ChangeEvents(POLLOUT);
 
-	boost::system_time const timeout = boost::get_system_time() + boost::posix_time::seconds(TLS_TIMEOUT_SECONDS);
+	boost::system_time const timeout = boost::get_system_time() + boost::posix_time::milliseconds(long(Configuration::TlsHandshakeTimeout * 1000));
```
Comment (N-o-X) [inline lib/base/tlsstream.cpp]: Why did you change this from seconds to milliseconds? I don't see the point.

Comment (dnsmichi) [reply lib/base/tlsstream.cpp]: Because the timeout can be configured as double, and otherwise you couldn't pass 1.555 for example.

Comment (N-o-X) [reply lib/base/tlsstream.cpp]: Seems legit 👍

**Comment by dnsmichi** [comment]
Added a short note to the upgrading docs.
